OsmIntegrator/Controllers/AccountController.cs

```diff
@@ -32,35 +32,31 @@ public class AccountController : Controller
   {
     private readonly SignInManager<ApplicationUser> _signInManager;
     private readonly UserManager<ApplicationUser> _userManager;
-    private readonly IEmailService _emailService;
     private readonly IConfiguration _configuration;
     private readonly ILogger<AccountController> _logger;
     private readonly RoleManager<ApplicationRole> _roleManager;
     private readonly ITokenHelper _tokenHelper;
     private readonly IStringLocalizer<AccountController> _localizer;
-    private readonly IExternalServicesConfiguration _externalServicesConfiguration;
+    private readonly IEmailHelper _emailHelper;
 
     public AccountController(
         UserManager<ApplicationUser> userManager,
         SignInManager<ApplicationUser> signInManager,
-        IEmailService emailService,
         IConfiguration configuration,
         ILogger<AccountController> logger,
         RoleManager<ApplicationRole> roleManager,
         ITokenHelper tokenHelper,
         IStringLocalizer<AccountController> localizer,
-        IExternalServicesConfiguration externalServicesConfiguration
-        )
+        IEmailHelper emailHelper)
     {
-      _logger = logger;
-      _userManager = userManager;
-      _signInManager = signInManager;
-      _emailService = emailService;
-      _configuration = configuration;
-      _roleManager = roleManager;
-      _tokenHelper = tokenHelper;
-      _localizer = localizer;
-      _externalServicesConfiguration = externalServicesConfiguration ?? throw new ArgumentNullException(nameof(externalServicesConfiguration));
+      _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+      _userManager = userManager ?? throw new ArgumentNullException(nameof(userManager));
```

Comment:
Warto to stosować?

Reply:
W przypadku obiektów dostarczanych przez IoC, to najczęściej używałem tego podejścia do zabezpieczenia przed popsuciem konfiguracji IoC. A w pozostałych przypadkach, to najzwyklejszy Guard sprawdzający, czy obiekt jest gotowa (w poprawnym stanie), zanim użyje się jakiejkolwiek jego metody/funkcji.